<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1020</width>
    <height>650</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Редактор кривых</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_2" columnstretch="1,0">
    <item row="0" column="0">
     <widget class="SGraph2dWidget" name="graph_widget" native="true"/>
    </item>
    <item row="0" column="1">
     <widget class="QGroupBox" name="groupBox">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>300</width>
        <height>0</height>
       </size>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="title">
       <string/>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <property name="leftMargin">
        <number>3</number>
       </property>
       <property name="topMargin">
        <number>3</number>
       </property>
       <property name="rightMargin">
        <number>3</number>
       </property>
       <property name="bottomMargin">
        <number>3</number>
       </property>
       <property name="horizontalSpacing">
        <number>3</number>
       </property>
       <item row="0" column="0" colspan="2">
        <widget class="QPointTableWidget" name="twPoints">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="layoutDirection">
          <enum>Qt::LeftToRight</enum>
         </property>
         <property name="horizontalScrollBarPolicy">
          <enum>Qt::ScrollBarAlwaysOff</enum>
         </property>
         <property name="editTriggers">
          <set>QAbstractItemView::DoubleClicked</set>
         </property>
         <property name="showDropIndicator" stdset="0">
          <bool>false</bool>
         </property>
         <property name="dragEnabled">
          <bool>false</bool>
         </property>
         <property name="dragDropOverwriteMode">
          <bool>false</bool>
         </property>
         <property name="dragDropMode">
          <enum>QAbstractItemView::NoDragDrop</enum>
         </property>
         <property name="alternatingRowColors">
          <bool>false</bool>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::SingleSelection</enum>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectItems</enum>
         </property>
         <property name="showGrid">
          <bool>true</bool>
         </property>
         <property name="cornerButtonEnabled">
          <bool>false</bool>
         </property>
         <property name="rowCount">
          <number>0</number>
         </property>
         <property name="columnCount">
          <number>2</number>
         </property>
         <attribute name="horizontalHeaderCascadingSectionResizes">
          <bool>false</bool>
         </attribute>
         <attribute name="horizontalHeaderDefaultSectionSize">
          <number>120</number>
         </attribute>
         <attribute name="horizontalHeaderHighlightSections">
          <bool>false</bool>
         </attribute>
         <attribute name="horizontalHeaderMinimumSectionSize">
          <number>120</number>
         </attribute>
         <attribute name="horizontalHeaderStretchLastSection">
          <bool>true</bool>
         </attribute>
         <attribute name="verticalHeaderVisible">
          <bool>false</bool>
         </attribute>
         <attribute name="verticalHeaderDefaultSectionSize">
          <number>24</number>
         </attribute>
         <attribute name="verticalHeaderHighlightSections">
          <bool>false</bool>
         </attribute>
         <attribute name="verticalHeaderMinimumSectionSize">
          <number>24</number>
         </attribute>
         <attribute name="verticalHeaderStretchLastSection">
          <bool>false</bool>
         </attribute>
         <column>
          <property name="text">
           <string>X</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Y</string>
          </property>
         </column>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QPushButton" name="pbDeletePoint">
         <property name="text">
          <string>Удалить точки</string>
         </property>
        </widget>
       </item>
       <item row="8" column="0">
        <widget class="QPushButton" name="pbPolygon">
         <property name="text">
          <string>Точка внутри многоугольника</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QCheckBox" name="cbOnFlyRebuild">
         <property name="text">
          <string>Рисовать на ходу</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="12" column="0">
        <widget class="QPushButton" name="pBCubicCurve">
         <property name="text">
          <string>Кубический сплайн</string>
         </property>
        </widget>
       </item>
       <item row="10" column="0">
        <widget class="QPushButton" name="pbTriangle">
         <property name="text">
          <string>Точка внутри треугольника</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QPushButton" name="pbNewControlLine">
         <property name="text">
          <string>Задать точки</string>
         </property>
        </widget>
       </item>
       <item row="12" column="1">
        <widget class="QCheckBox" name="cBDeriv">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Производные / Веса</string>
         </property>
        </widget>
       </item>
       <item row="14" column="0">
        <widget class="QPushButton" name="pbHermitCurve">
         <property name="text">
          <string>Сплайн Эрмита</string>
         </property>
        </widget>
       </item>
       <item row="8" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <property name="spacing">
          <number>6</number>
         </property>
         <item>
          <widget class="QLabel" name="label">
           <property name="layoutDirection">
            <enum>Qt::LeftToRight</enum>
           </property>
           <property name="text">
            <string>Вершины</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="sBpolygonCount">
           <property name="focusPolicy">
            <enum>Qt::NoFocus</enum>
           </property>
           <property name="contextMenuPolicy">
            <enum>Qt::NoContextMenu</enum>
           </property>
           <property name="toolTip">
            <string>Кол-во вершин</string>
           </property>
           <property name="toolTipDuration">
            <number>0</number>
           </property>
           <property name="whatsThis">
            <string/>
           </property>
           <property name="readOnly">
            <bool>false</bool>
           </property>
           <property name="specialValueText">
            <string notr="true"/>
           </property>
           <property name="suffix">
            <string/>
           </property>
           <property name="prefix">
            <string/>
           </property>
           <property name="minimum">
            <number>3</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="2" column="1">
        <widget class="QCheckBox" name="cbShowBroken">
         <property name="text">
          <string>Показывать ломаную</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="15" column="0">
        <widget class="QPushButton" name="pbLagrange">
         <property name="text">
          <string>Сплайн Лагранжа</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QPushButton" name="pBevolute">
         <property name="text">
          <string>Эволюта</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QPushButton" name="pbDelaunay">
         <property name="text">
          <string>Триангуляция Делоне</string>
         </property>
        </widget>
       </item>
       <item row="11" column="0">
        <widget class="QPushButton" name="pbRotate">
         <property name="text">
          <string>Предикат поворота</string>
         </property>
        </widget>
       </item>
       <item row="11" column="1">
        <widget class="QPushButton" name="pBrationalBezier">
         <property name="text">
          <string>Рац. кривая Безье</string>
         </property>
        </widget>
       </item>
       <item row="15" column="1">
        <widget class="QPushButton" name="phHermitEnergy">
         <property name="text">
          <string>Псевдоупругий</string>
         </property>
        </widget>
       </item>
       <item row="14" column="1">
        <widget class="QPushButton" name="pbBezierCurve">
         <property name="text">
          <string>Кривая Безье</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>SGraph2dWidget</class>
   <extends>QWidget</extends>
   <header>sgraph2dwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QPointTableWidget</class>
   <extends>QTableWidget</extends>
   <header>qpointtablewidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
